### PR TITLE
Fixes #10072: an 'admin' user with sudo rights is now being configured for cloud-init setups.

### PR DESCRIPTION
**Pull request by dmitri-d**
This allows for spcification of non-clear text password for root-like administrative account. Part of support for atomic vm provisioning.

### CODE REVIEW

File: cloudinit/userdata_cloudinit.erb [hunk 1]
```diff
@@ -21,8 +24,21 @@ hostname: <%= @host.shortname %>
...
+lock-passwd: false
+passwd: <%= @host.root_pass %>
+
 <%# Contact Foreman to confirm instance is built -%>
 phone_home:
- url: <%= foreman_url('built') %>
- post: []
- tries: 10
+url: <%= foreman_url('built') %>
+post: []
```
Comment (domcleal) [inline cloudinit/userdata_cloudinit.erb]: seems many lines have lost their YAML hash indentation?

**Comment by dmitri-d** [comment]
@domcleal: fixed!

**Comment by ekohl** [comment]
I wonder if this should be hidden behind an if.

**Comment by dmitri-d** [comment]
Not sure what could be used for a flag. I think I'd rather have a dedicated template rather then a flag/conditional check if this isn't needed in general.

**Comment by domcleal** [comment]
We'd normally use conditionals in templates so we don't have to seed so many.  If it's specifically needed for Atomic somehow, then put it behind a check for the OS name?

**Comment by dmitri-d** [comment]
With cloud-init the issue is that if one needs to change the password on an existing account (including root or root-like ones), it will have to be specified in clear-text. This is not distribution-specific, but probably more influenced by usage scenarios, hence my question. 

Having said that, the old root (or `centos`, `fedora`, or what have you) account is still present and available, this only creates an additional account.

**Comment by ares** [comment]
@witlessbird could you please reopen against develop branch? we renamed master to develop since we started branching this repo per foreman releases

**Pull request closed without merging**